actionpack/lib/action_dispatch/http/content_security_policy.rb

```diff
@@ -175,13 +175,7 @@ def build(context = nil)
 
     private
       def copy_directives(directives)
-        directives.transform_values do |sources|
-          if sources.is_a?(Array)
-            sources.map(&:dup)
-          else
-            sources
-          end
-        end
+        directives.transform_values { |sources| sources.deep_dup }
```

Comment:
Can do `directives.deep_dup` directly. At which point I'd just get rid of the method and use `#deep_dup` directly in `#initialize_copy`:
```ruby
def initialize_copy(other)
  @directives = other.directives.deep_dup
end
```